odoo/addons/base/models/res_partner.py

```diff
@@ -979,9 +979,12 @@ def address_get(self, adr_pref=None):
                         result[record.type] = record.id
                     if len(result) == len(adr_pref):
                         return result
-                    to_scan = [c for c in record.child_ids
-                                 if c not in visited
-                                 if not c.is_company] + to_scan
+                    domain = [("id", "not in", [i.id for i in visited]),
+                              ("parent_id", "=", record.id), ("is_company", "!=", True)]
```

Comment:
Double negation could be avoided here:

```suggestion
                              ("parent_id", "=", record.id), ("is_company", "=", False)]
```

Reply:
Odoo didn't consider Null case for positive operators

But I just tested both domains and they are generating the same query

Thank you!